src/core/filesystem.cc

```diff
@@ -1042,7 +1042,7 @@ GetFileSystem(const std::string& path, FileSystem** file_system)
         "-DTRTIS_ENABLE_S3=ON.");
 #else
     re2::RE2 s3_regex(
```

Comment:
You can make the regex to be a class member, seems to be identical as the one above.

Reply:
Done